python/cudf/cudf/_lib/cpp/interop.pxd

```diff
@@ -25,7 +25,13 @@ cdef extern from "cudf/interop.hpp" namespace "cudf" \
 
     cdef unique_ptr[table] from_arrow(CTable input) except +
 
+    cdef cppclass column_metadata:
```

Comment:
This looks pretty much identical to the `column_name_info` data structure being used in cuIO: https://github.com/rapidsai/cudf/blob/branch-0.16/cpp/include/cudf/io/types.hpp#L98

Can that just be reused in both the C++ and Cython layer? cc: @nvdbaranec @kkraus14

Reply:
I thought of keeping it different as in future if there is any new metadata that might be required to be added just in case of Arrow would be easier. But, we can make it common.

Reply:
+1 to sharing this if possible

Reply:
For what it's worth, we are probably going to revisit the column_name_info and sibling column_names fields in the future.   

column_name_info is a superset of simple column_names, but represents the effective names of the cudf output columns,  _not_ the names in the parquet schema itself.  Whether or not this is a Good Thing or not is still up in the air.  

Anyway, there may be some delta here in the future.